x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/dataframe/MappingsMergerTests.java

```diff
@@ -38,7 +38,12 @@ public void testMergeMappings_GivenIndicesWithIdenticalMappings() {
 
         MappingMetadata mergedMappings = MappingsMerger.mergeMappings(newSource(), getMappingsResponse);
 
-        assertThat(mergedMappings.getSourceAsMap(), equalTo(index1Mappings));
+        Map<String, Object> mergedMappingsMap = mergedMappings.getSourceAsMap();
+        assertThat(mergedMappingsMap.size(), equalTo(2));
+        assertThat(mergedMappingsMap.containsKey("dynamic"), is(true));
+        assertThat(mergedMappingsMap.get("dynamic"), equalTo(false));
+        assertThat(mergedMappingsMap.containsKey("properties"), is(true));
+        assertThat(mergedMappingsMap.get("properties"), equalTo(index1Mappings.get("properties")));
```

Comment:
If you constructed a map containing the expected contents, then asserted equality with the merged mappings map then you'd get better feedback in the case of failure - both maps would get printed instead of (for example) just finding out that there were 3 items instead of 2.